### PR TITLE
Prop types

### LINKED ISSUE #319: Properties mapping
### Description

Make sure the new properties work. Update Catalog app so that each scene uses new API and behaves as it used to.

### Completion status
- [ ] [Android](https://github.com/magic-script/magic-script-components-react-native/issues/607)
- [ ] [iOS](https://github.com/magic-script/magic-script-components-react-native/issues/607)
- [ ] [Lumin]()

### PR DESCRIPTION
**Pull request by pleszkiewicz**
Update PropTypes for all components.

Closes #319 

---
Thank you for your contribution to MagicScript Project.
Before submitting this PR, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests
- [ ] You have updated documentation

### CODE REVIEW

File: src/components/web-view.js [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-  return React.createElement("webView", mapProperties("WebView", props));
+  return React.createElement('webView', mapProperties('WebView', props));
 }
 
 WebView.propTypes = {
-  ...UiNode.PropTypes
+  ...UiNode.PropTypes,
+
+  // Properties
+  action: PropTypes.oneOf(Object.values(WebViewAction)),
```
Comment (kpiascik) [inline src/components/web-view.js:16]: how is `action` supposed to be used?  We're not using it the Catalog app.  I'm not sure if it makes sense.

File: src/components/portal-icon.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import React from "react";
-import mapProperties from "../properties-mapping.js";
+import React from 'react';
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (pleszkiewicz) [reply src/components/portal-icon.js:2]: PropTypes are not used in here. I can remove this line but should we support Portal icon in Lumin (it's not supported by mobile)?

File: src/components/prism.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  onRotationChanged: PropTypes.func,
+  onScaleChanged: PropTypes.func,
+  onPositionChanged: PropTypes.func
+};
+
+Prism.defaultProps = {
+  position: [0, 0, 0],
+  orientation: [0, 0, 0, 1],
+  scale: [1, 1, 1],
+  size: [1, 1, 1]
```
[the remark this answers is not in the document]
Comment (pleszkiewicz) [reply src/components/prism.js:29]: Hmmm, I wonder if we need to add a default value for `size`? `size` is set as required so the user will have to provide it anyway. I'm going to remove the default value.

File: src/components/portal-icon.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
+import mapProperties from '../properties-mapping.js';
 
 export function PortalIcon(props) {
-  return React.createElement("portalIcon", mapProperties("PortalIcon", props));
+  return React.createElement('portalIcon', mapProperties('PortalIcon', props));
 }
 
 PortalIcon.propTypes = {
```
[the remark this answers is not in the document]
Comment (pleszkiewicz) [reply src/components/portal-icon.js:9]:
```
class INTERGEN_EXPORT UiPortalIcon : public UiNode
```

File: src/components/spinner.js [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
 
 Spinner.propTypes = {
   ...UiNode.PropTypes,
 
   width: PropTypes.number,
   height: PropTypes.number,
-  value: PropTypes.number,
+  value: TypeValidator.range(0, 1),
   type: PropTypes.oneOf(Object.values(SpinnerType)),
-  path: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (pleszkiewicz) [reply src/components/spinner.js:19]:
```
@param resPath - The resource path in the resource (optional).
```
We have never used this param and no one will ever use it :)

File: src/components/web-view.js (hunk 1 above)
Comment (pleszkiewicz) [reply src/components/web-view.js:16]: Should we remove `action` from here and from the documentation?

**Review by kpiascik — APPROVED** [review]

File: src/components/web-view.js (hunk 1 above)
Comment (kpiascik) [reply src/components/web-view.js:16]: I think we should remove it

File: src/components/portal-icon.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpiascik) [reply src/components/portal-icon.js]: I think we should remove the portal-icon

**Pull request merged**